x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/extractor/scroll/ScrollDataExtractor.java

```diff
@@ -214,12 +221,14 @@ void markScrollAsErrored() {
 
     @SuppressWarnings("HiddenField")
     protected SearchResponse executeSearchScrollRequest(String scrollId) {
-        return ClientHelper.executeWithHeaders(
+        SearchResponse searchResponse = ClientHelper.executeWithHeaders(
             context.headers,
             ClientHelper.ML_ORIGIN,
             client,
             () -> new SearchScrollRequestBuilder(client, SearchScrollAction.INSTANCE).setScroll(SCROLL_TIMEOUT).setScrollId(scrollId).get()
         );
+        checkForSkippedClusters(searchResponse);
```

Comment:
We don't need to check this for subsequent scroll requests, but it's okay to leave this check as is here.